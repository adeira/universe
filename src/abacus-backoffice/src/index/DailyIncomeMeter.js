// @flow

import { graphql, useFragment } from '@adeira/relay';
import { LayoutBlock, Meter, Money } from '@adeira/sx-design';
import type { Node } from 'react';

import refineSupportedCurrencies from '../refineSupportedCurrencies';
import type { DailyIncomeMeterFragment$key } from './__generated__/DailyIncomeMeterFragment.graphql';

type Props = {
  +data: DailyIncomeMeterFragment$key,
};

export default function DailyIncomeMeter(props: Props): Node {
  const dailyReport = useFragment(
    graphql`
      fragment DailyIncomeMeterFragment on AnalyticsDailyReportInfo {
        total {
          unitAmount
          unitAmountCurrency
        }
      }
    `,
    props.data,
  );

  const dailyGoal = 2500; // TODO: MXN only

  return (
    <LayoutBlock>
      <Meter
        min={0}
        max={dailyGoal}
        value={
          dailyReport.total.unitAmount / 100 // adjusted for centavo
        }
        low={dailyGoal * 0.8}
        optimum={dailyGoal}
      />
      <Money
        priceUnitAmount={
          dailyReport.total.unitAmount / 100 // adjusted for centavo
        }
        priceUnitAmountCurrency={refineSupportedCurrencies(dailyReport.total.unitAmountCurrency)}
      />
    </LayoutBlock>
  );
}
