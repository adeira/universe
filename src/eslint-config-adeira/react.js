// @flow

const { react } = require('./presetsConfig');
const changeNextVersionErrorLevel = require('./changeNextVersionErrorLevel');
const getCommonConfig = require('./getCommonConfig');

/*::

import type { EslintConfig } from './EslintConfig.flow';

*/

const WARN = 1;

const preset = ({
  ...getCommonConfig(changeNextVersionErrorLevel(react.rules, WARN)),
  plugins: react.plugins,
} /*: EslintConfig */);

module.exports = preset;
