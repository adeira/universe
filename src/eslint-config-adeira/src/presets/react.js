// @flow

const { ERROR, OFF, WARN } = require('../constants');
const detectReactVersion = require('../detectReactVersion');

/*::

import type { EslintConfig } from '../EslintConfig.flow';

*/

module.exports = ({
  plugins: [
    'eslint-plugin-react',
    'eslint-plugin-react-hooks',
    'eslint-plugin-react-native',
    'eslint-plugin-jsx-a11y',
  ],
  rules: {
    // React (https://github.com/yannickcr/eslint-plugin-react)
    'react/boolean-prop-naming': OFF,
    'react/button-has-type': [ERROR, { button: true, submit: true, reset: false }],
    'react/default-props-match-prop-types': OFF,
    'react/destructuring-assignment': OFF,
    'react/display-name': OFF,
    'react/forbid-component-props': OFF,
    'react/forbid-dom-props': [
      ERROR,
      {
        forbid: [
          'class', // https://reactjs.org/docs/dom-elements.html#classname
          'for', // https://reactjs.org/docs/dom-elements.html#htmlfor
        ],
      },
    ],
    'react/forbid-elements': OFF,
    'react/forbid-foreign-prop-types': OFF,
    'react/forbid-prop-types': OFF,
    'react/jsx-boolean-value': OFF,
    'react/jsx-curly-brace-presence': OFF,
    'react/jsx-filename-extension': OFF,
    'react/jsx-fragments': OFF,
    'react/jsx-handler-names': OFF,
    'react/jsx-key': ERROR,
    'react/jsx-max-depth': OFF,
    'react/jsx-no-bind': [
      ERROR,
      {
        ignoreDOMComponents: true,
        ignoreRefs: true,
        allowArrowFunctions: true,
        allowFunctions: false,
        allowBind: false,
      },
    ],
    'react/jsx-no-comment-textnodes': ERROR,
    'react/jsx-no-constructed-context-values': ERROR,
    'react/jsx-no-duplicate-props': ERROR,
    'react/jsx-no-literals': OFF,
    'react/jsx-no-target-blank': ERROR,
    'react/jsx-no-undef': ERROR,
    'react/jsx-no-useless-fragment': ERROR,
    'react/jsx-pascal-case': ERROR,
    'react/jsx-props-no-spreading': OFF,
    'react/jsx-sort-default-props': OFF,
    'react/jsx-sort-props': OFF,
    'react/jsx-uses-react': WARN,
    'react/jsx-uses-vars': ERROR,
    'react/no-access-state-in-setstate': ERROR,
    'react/no-array-index-key': OFF,
    'react/no-children-prop': OFF,
    'react/no-danger': OFF,
    'react/no-danger-with-children': ERROR,
    'react/no-deprecated': ERROR,
    'react/no-did-mount-set-state': OFF, // https://github.com/airbnb/javascript/issues/684#issuecomment-264094930
    'react/no-did-update-set-state': ERROR,
    'react/no-direct-mutation-state': ERROR,
    'react/no-find-dom-node': ERROR,
    'react/no-is-mounted': ERROR,
    'react/no-multi-comp': [ERROR, { ignoreStateless: true }],
    'react/no-redundant-should-component-update': ERROR,
    'react/no-render-return-value': ERROR,
    'react/no-set-state': OFF,
    'react/no-string-refs': OFF,
    'react/no-this-in-sfc': ERROR,
    'react/no-typos': OFF,
    'react/no-unescaped-entities': ERROR,
    'react/no-unknown-property': ERROR,
    'react/no-unsafe': OFF,
    'react/no-unstable-nested-components': [ERROR, { allowAsProps: true }],
    'react/no-unused-prop-types': OFF,
    'react/no-unused-state': OFF, // Enable when they fix this issue: https://github.com/yannickcr/eslint-plugin-react/issues/1910
    'react/no-will-update-set-state': OFF,
    'react/prefer-es6-class': [ERROR, 'always'],
    'react/prefer-exact-props': OFF, // see: fb-flow/use-exact-by-default-object-type
    'react/prefer-read-only-props': OFF,
    'react/prefer-stateless-function': OFF,
    'react/prop-types': OFF, // we use Flow instead,
    'react/react-in-jsx-scope': OFF,
    'react/require-default-props': OFF,
    'react/require-optimization': OFF,
    'react/require-render-return': ERROR,
    'react/self-closing-comp': ERROR,
    'react/sort-comp': [
      ERROR,
      {
        // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/sort-comp.md
        order: [
          'type-annotations',
          'static-variables',
          'static-methods',
          'instance-variables',
          'lifecycle',
          'everything-else',
          'rendering',
        ],
        groups: {
          lifecycle: [
            'displayName',
            'propTypes',
            'contextTypes',
            'childContextTypes',
            'mixins',
            'statics',
            'defaultProps',
            'constructor',
            'getDefaultProps',
            'state',
            'getInitialState',
            'getChildContext',
            'getDerivedStateFromProps', // this is added
            'getDerivedStateFromError', // this is added
            'componentWillMount',
            'UNSAFE_componentWillMount',
            'componentDidMount',
            'componentWillReceiveProps',
            'UNSAFE_componentWillReceiveProps',
            'shouldComponentUpdate',
            'componentWillUpdate',
            'UNSAFE_componentWillUpdate',
            'getSnapshotBeforeUpdate',
            'componentDidUpdate',
            'componentDidCatch',
            'componentWillUnmount',
          ],
          rendering: ['/^render.+$/', 'render'],
        },
      },
    ],
    'react/sort-prop-types': OFF,
    'react/state-in-constructor': OFF,
    'react/static-property-placement': OFF,
    'react/style-prop-object': ERROR,
    'react/void-dom-elements-no-children': ERROR,
    'react/function-component-definition': OFF,
    'react/jsx-no-script-url': ERROR,
    'react/no-adjacent-inline-elements': OFF,

    // React Hooks (https://www.npmjs.com/package/eslint-plugin-react-hooks)
    'react-hooks/rules-of-hooks': ERROR,
    'react-hooks/exhaustive-deps': [
      ERROR,
      {
        // https://recoiljs.org/docs/introduction/installation/#eslint
        additionalHooks: 'useRecoilCallback', // should be regex!
      },
    ],

    // React Native (https://github.com/intellicode/eslint-plugin-react-native)
    'react-native/no-color-literals': OFF,
    'react-native/no-inline-styles': OFF,
    'react-native/no-raw-text': OFF, // TODO: takes normal React into account as well, not only RN
    'react-native/no-single-element-style-arrays': ERROR,
    'react-native/no-unused-styles': ERROR,
    'react-native/sort-styles': OFF,
    'react-native/split-platform-components': OFF,

    // React Accessibility (https://github.com/evcohen/eslint-plugin-jsx-a11y)
    'jsx-a11y/accessible-emoji': ERROR,
    'jsx-a11y/alt-text': ERROR,
    'jsx-a11y/anchor-has-content': ERROR,
    'jsx-a11y/anchor-is-valid': ERROR,
    'jsx-a11y/aria-activedescendant-has-tabindex': OFF,
    'jsx-a11y/aria-props': ERROR,
    'jsx-a11y/aria-proptypes': ERROR,
    'jsx-a11y/aria-role': ERROR,
    'jsx-a11y/aria-unsupported-elements': ERROR,
    'jsx-a11y/autocomplete-valid': ERROR,
    'jsx-a11y/click-events-have-key-events': OFF,
    'jsx-a11y/control-has-associated-label': OFF,
    'jsx-a11y/heading-has-content': ERROR,
    'jsx-a11y/html-has-lang': ERROR,
    'jsx-a11y/iframe-has-title': ERROR,
    'jsx-a11y/img-redundant-alt': OFF,
    'jsx-a11y/interactive-supports-focus': ERROR,
    'jsx-a11y/label-has-associated-control': OFF,
    'jsx-a11y/label-has-for': OFF,
    'jsx-a11y/lang': ERROR,
    'jsx-a11y/media-has-caption': OFF,
    'jsx-a11y/mouse-events-have-key-events': OFF,
    'jsx-a11y/no-access-key': ERROR,
    'jsx-a11y/no-autofocus': [ERROR, { ignoreNonDOM: true }],
    'jsx-a11y/no-distracting-elements': [ERROR, { elements: ['marquee', 'blink'] }],
    'jsx-a11y/no-interactive-element-to-noninteractive-role': OFF,
    'jsx-a11y/no-noninteractive-element-interactions': OFF,
    'jsx-a11y/no-noninteractive-element-to-interactive-role': OFF,
    'jsx-a11y/no-noninteractive-tabindex': ERROR,
    'jsx-a11y/no-onchange': OFF,
    'jsx-a11y/no-redundant-roles': ERROR,
    'jsx-a11y/no-static-element-interactions': OFF,
    'jsx-a11y/role-has-required-aria-props': ERROR,
    'jsx-a11y/role-supports-aria-props': ERROR,
    'jsx-a11y/scope': ERROR,
    'jsx-a11y/tabindex-no-positive': ERROR,
  },
  settings: {
    react: {
      version: detectReactVersion(),
    },
  },
} /*: EslintConfig */);
