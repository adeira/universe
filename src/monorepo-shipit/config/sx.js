// @flow strict

import type { ConfigType } from '../ConfigType.flow';

module.exports = ({
  getStaticConfig() {
    return {
      repository: 'git@github.com:adeira/sx.git',
    };
  },
  getPathMappings() {
    return new Map([['src/sx/', '']]);
  },
}: ConfigType);
