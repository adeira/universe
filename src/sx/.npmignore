__flowtests__
__tests__
