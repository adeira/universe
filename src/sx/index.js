/**
 * Users of SX should be able to import it as follows:
 *
 *    import sx from '@adeira/sx';
 *    sx(), sx.create(), sx.keyframes(), ...
 *
 * Alternatively:
 *
 *    import { create as sxCreate } from '@adeira/sx';
 *    sxCreate(), ...
 *
 * @flow
 */

import create, { type AllCSSProperties } from './src/create';
import keyframes from './src/keyframes';
import renderPageWithSX from './src/renderPageWithSX';

/**
 * This function allows us to compose local and external styles like so:
 *
 * ```
 * import sx from '@adeira/sx';
 *
 * const styles = sx.create({ default: { fontSize: 16 } });
 * const externalStyles = sx.create({ spacing: { marginTop: 4 } });
 *
 * <div className={sx(styles.default, externalStyles.spacing)} />;
 * ```
 *
 * It does so by literally merging the 2 objects together and calling `sx.create` on the result.
 */
function composeStylesheets(...stylesheets: $ReadOnlyArray<?AllCSSProperties>): ?string {
  // Should we support deeply nested styles or leave it like this and overwrite them?
  const mergedStylesheet = Object.assign({}, ...stylesheets);
  if (Object.keys(mergedStylesheet).length === 0) {
    // happens when composing "nothing" which is a valid input
    return undefined;
  }
  return create({
    __internal_compose_stylesheets: mergedStylesheet,
  })('__internal_compose_stylesheets');
}

composeStylesheets.create = create;
composeStylesheets.keyframes = keyframes;
composeStylesheets.renderPageWithSX = renderPageWithSX;

export { create, keyframes, renderPageWithSX };
export default composeStylesheets;
export type { AllCSSProperties } from './src/create';
