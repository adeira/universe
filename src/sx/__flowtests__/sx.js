// @flow

import React, { type Node } from 'react';

import * as sx from '../index';

export function NoErrors(): Node {
  const styles = sx.create({
    red: { color: 'red' },
  });
  return <div className={styles('red')} />;
}

export function NoErrorsWithPseudo(): Node {
  const styles = sx.create({
    red: {
      'color': 'red',
      ':hover': {
        color: 'blue',
      },
      '::after': {
        content: '"â˜…"',
      },
    },
  });
  return <div className={styles('red')}>I am blue on hover</div>;
}

export function NoErrorsMoreStylesheetNames(): Node {
  const styles = sx.create({
    red: { color: 'red' },
    blue: { color: 'blue' },
  });
  return <div className={styles('red', 'blue')} />;
}

export function InvalidStylesheetName(): Node {
  const styles = sx.create({
    red: { color: 'red' },
  });
  // $FlowExpectedError[incompatible-call]
  return <div className={styles('invalid')} />;
}

export function InvalidCSSProperty(): Node {
  const styles = sx.create({
    red: {
      // $FlowExpectedError[incompatible-call]
      wtf: 'wtf',
    },
  });
  // $FlowExpectedError[incompatible-call]
  return <div className={styles('invalid')} />;
}

export function InvalidCSSPseudoProperty(): Node {
  const styles = sx.create({
    red: {
      // $FlowExpectedError[incompatible-call]
      ':hover': -1,
    },
  });
  return <div className={styles('red')} />;
}

export function FlexNonNumericValues(): Node {
  const styles = sx.create({
    'flex-1': { flex: '1 1 0%' },
    'flex-auto': { flex: '1 1 auto' },
    'flex-initial': { flex: '0 1 auto' },
    'flex-none': { flex: 'none' },
  });
  return <div className={styles('flex-1', 'flex-auto', 'flex-initial', 'flex-none')} />;
}

export function GridColumnNonNumericValues(): Node {
  const styles = sx.create({
    'col-auto': { gridColumn: 'auto' },
    'col-end-auto': { gridColumnEnd: 'auto' },
    'col-span-1': { gridColumn: 'span 1 / span 1' },
  });
  return <div className={styles('col-auto', 'col-end-auto', 'col-span-1')} />;
}

export function GridRowNonNumericValues(): Node {
  const styles = sx.create({
    'row-auto': { gridRow: 'auto' },
    'row-end-auto': { gridRowEnd: 'auto' },
    'row-span-1': { gridRow: 'span 1 / span 1' },
    'row-start-auto': { gridRowStart: 'auto' },
  });
  return <div className={styles('row-auto', 'row-end-auto', 'row-span-1', 'row-start-auto')} />;
}

export function ZIndex(): Node {
  const styles = sx.create({
    'z-auto': { zIndex: 'auto' },
    'z-10': { zIndex: 10 },
  });
  return <div className={styles('z-auto', 'z-10')} />;
}
