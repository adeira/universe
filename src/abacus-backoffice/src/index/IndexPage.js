// @flow

import * as React from 'react';
import { Note } from '@adeira/sx-design';
import { graphql, useLazyLoadQuery } from '@adeira/relay';

import type { IndexPageQuery } from './__generated__/IndexPageQuery.graphql';

export default function IndexPage(): React.Node {
  const data = useLazyLoadQuery<IndexPageQuery>(
    /* eslint-disable relay/unused-fields */
    graphql`
      query IndexPageQuery {
        whoami {
          id
          humanReadableType
        }
      }
    `,
    /* eslint-enable relay/unused-fields */
  );

  return (
    <>
      <Note tint="warning">work in progress</Note>
      <pre>{JSON.stringify(data, null, 2)}</pre>
    </>
  );
}
