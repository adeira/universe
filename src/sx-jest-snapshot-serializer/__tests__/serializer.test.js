// @flow

import sx from '@adeira/sx';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import '../index'; // the serializer itself

function TestComponent() {
  const styles = sx.create({ test: { color: 'red' } });
  return <div className={styles('test')}>test component</div>;
}

it('works as expected with "react-test-renderer"', () => {
  const testRenderer = TestRenderer.create(<TestComponent />);
  expect(testRenderer.toJSON()).toMatchSnapshot();
});

it('works as expected "@testing-library/react"', () => {
  const { container } = render(<TestComponent />);
  expect(container).toMatchSnapshot();
});
