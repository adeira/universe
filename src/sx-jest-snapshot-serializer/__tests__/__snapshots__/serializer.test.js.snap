// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works as expected "@testing-library/react" 1`] = `
._324Crd {
  color: #f00;
}

<div>
  <div class="_324Crd">test component</div>
</div>
`;

exports[`works as expected with "react-test-renderer" 1`] = `
._324Crd {
  color: #f00;
}

<div className="_324Crd">test component</div>
`;
